lib/weather/accounts/accounts.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Weather.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Weather.Repo
+
+  alias Weather.Accounts.User
+
+  @doc """
+  Returns the list of users.
+  """
+  def list_users do
+    Repo.all(User)
+  end
+
+  @doc """
+  Gets a single user.
+  """
+  def find_user(id), do: Repo.get(User, id)
+
+  @doc """
+  Gets a single user.
+  """
+  def find_user_by_email(email) do
+    User
+    |> Repo.get_by(email: email)
+  end
+
+  @doc """
+  Authenticate a user.
+  """
+  def authenticate_user(email, password) do
+    user = find_user_by_email(String.downcase(email))
+    with {:error, _} <- Comeonin.Pbkdf2.check_pass(user, password) do
```

Comment:
You should get the first line in your `with` and handle the successful path in the `do` block, then pattern match errors in the `else` block.